### PR TITLE
Better group management error handling, backend side #9055

### LINKED ISSUE #9055: group creation errors not shown
@rgdonohue was trying to create a new group with a space in the name, but the new group wasn't created (or at least listed in the UI), and no error was thrown.

Steps to reproduce (:

1. Click "Create new Group"
2. Name the group something like "Fancy new group"
3. Click create group
4. The UI puts you back to the group screen without the new group being listed

It seems that either there should be an error about having a space in the new group, or groups should allow spaces in their names. I support the second option.

cc @juanignaciosl

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #9055

@xavijam , @viddo, @nobuti , I'd need help with frontend, ping me, please.

### CODE REVIEW

File: lib/assets/javascripts/cartodb/organization/groups_admin/create_group_view.js [hunk 1]
```diff
@@ -62,10 +62,12 @@ module.exports = cdb.core.View.extend({
 
   _showErrors: function() {
     this.model.set('isLoading', false);
+    // TODO: display response errors
   },
 
   _onChangeName: function() {
-    this.$('.js-create').toggleClass('is-disabled', this._name().length === 0);
+    var disabled = this._name().length === 0 || this._name().length > 28;
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/organization/groups_admin/create_group_view.js]: I think it's a better idea to set a max limit on the input field.

**Comment by viddo** [comment]
What do you need help with?

**Comment by juanignaciosl** [comment]
I'd like create group view to display errors from backend. I haven't been able to find whether errors can be currently displayed, or how. Backend now returns `{ errors: "ERROR MESSAGE" }`. Even if a length validation limit is included in the frontend, back might fail because of other reasons, and now it displays nothing.

**Comment by viddo** [comment]
I'll have a look…

**Comment by juanignaciosl** [comment]
Thanks, @viddo . I took a look but I couldn't find _the_ way to do it. If you point me to an example I can follow I'll do it.

**Comment by viddo** [comment]
There is really no "_the_" way to do this (yet), I'll implement a simple solution to solve the current needs

**Comment by viddo** [comment]
@juanignaciosl See the last 2 commits, e.g.:
screen shot 2016-07-26 at 14 48 50

Reusing the existing flash-message header but having an implementation that can be reused for other places where we want to have the general error feedback. I added it to edit-group UI too.

**Comment by juanignaciosl** [comment]
@viddo works like a charm :-) About the question, most frequent PG errors pattern is matched, extracting the meaningful part (although this will be improved as more errors appear). Thanks!

@javitonino can you CR the backend part and @xavijam the frontend? Thanks!

File: app/controllers/carto/api/groups_controller.rb [hunk 2]
```diff
@@ -136,6 +146,16 @@ def load_organization_users
         render json: { errors: "Users #{ids} not found" }, status: 404 unless @organization_users.length > 0
       end
 
+      def handle_statement_invalid_error(e, group)
+        err_regexp = /ERROR:  (.*)\n/
+        if e.message =~ err_regexp
+          render json: { errors: err_regexp.match(e.message)[1] }, status: 422
+        else
+          CartoDB.notify_exception(e, params: params, group: group ? group : 'not created', organization: @organization)
```
Comment (javitonino) [inline app/controllers/carto/api/groups_controller.rb]: Please, use `CartoDB::Logger`.

File: app/controllers/carto/api/groups_controller.rb [hunk 3]
```diff
@@ -136,6 +146,16 @@ def load_organization_users
         render json: { errors: "Users #{ids} not found" }, status: 404 unless @organization_users.length > 0
       end
 
+      def handle_statement_invalid_error(e, group)
+        err_regexp = /ERROR:  (.*)\n/
+        if e.message =~ err_regexp
+          render json: { errors: err_regexp.match(e.message)[1] }, status: 422
```
Comment (javitonino) [inline app/controllers/carto/api/groups_controller.rb]: Minor: we usually return an array or hash of errors from these things. Probably not worth it to change it for an internal API.

**Comment by javitonino** [comment]
LGTM. Left a couple of minor comments, but feel free to disregard them.

File: app/controllers/carto/api/groups_controller.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/controllers/carto/api/groups_controller.rb]: Sure, I always forget :_)

File: app/controllers/carto/api/groups_controller.rb (hunk 3 above)
Comment (juanignaciosl) [inline app/controllers/carto/api/groups_controller.rb]: I'll give it a try, forcing me to review all @viddo frontend code.

Comment (viddo) [inline app/controllers/carto/api/groups_controller.rb]: The only issue would be formatting really, an array with multiple items would be converted to string using the default concatenation, e.g.:
```js
['first', 'second', 'third'].toString() // => "first,second,third"
```

Could handle it a little more gracefully if you want to?

**Comment by juanignaciosl** [comment]
@viddo I've solved the message joining at group view level, as you can see in my last commit. In the beginning I thought about solving it at flash message level but making a flash message not a single message but an array felt like engineering. What do you think?

@javitonino you can CR backend changes.

**Comment by javitonino** [comment]
Oh, I didn't realize about so many other error messages when suggesting making the response an array. Sorry for that. Backend: :sparkles:

**Comment by viddo** [comment]
@juanignaciosl 👍

**Comment by viddo** [comment]
@juanignaciosl my last commit should fix the failing specs too

**Comment by juanignaciosl** [comment]
@viddo ❤️  thanks! I had just begun with it :-)

**Pull request merged**